### PR TITLE
Create invoice

### PR DESCRIPTION
**Pull request by makaeron**
- [x] #10 
- [x] #5
- [x] #13

### CODE REVIEW

File: src/main/java/com/chaos/invoicify/dto/InvoiceDto.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import java.util.List;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class InvoiceDto {
+    String invoiceName;
+    String companyName;
+    String invoiceDate;
```
Comment (samik-saha) [inline src/main/java/com/chaos/invoicify/dto/InvoiceDto.java]: Replace invoiceName with invoiceNumber. Add totalCost and modifiedDate.
Replace invoiceDate with createDate. createDate and modifiedDate should be of Date type

File: src/main/java/com/chaos/invoicify/entity/InvoiceEntity.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    Long id;
+
+    @OneToMany(fetch = FetchType.LAZY,mappedBy = "invoice")
+    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
+    List<ItemEntity> items;
+
+    String invoiceName;
+    String companyName;
```
Comment (samik-saha) [inline src/main/java/com/chaos/invoicify/entity/InvoiceEntity.java]: Company name should be replaced with real CompanyEntity type

File: src/main/java/com/chaos/invoicify/service/InvoiceService.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    public List<ItemDto> fetchAllItems(InvoiceEntity invoiceEntity) {
+        return itemRepository.findAll()
+                .stream()
+                .filter(itemEntity -> itemEntity.getInvoice().equals(invoiceEntity))
+                .map(itemEntity -> {
+                    return new ItemDto(itemEntity.getItemDescription(),
+                            itemEntity.getItemCount(),
+                            itemEntity.getItemFeeType(),
+                            itemEntity.getItemUnitPrice()
+//                          itemEntity.getInvoice().getInvoiceName()
```
Comment (samik-saha) [inline src/main/java/com/chaos/invoicify/service/InvoiceService.java]: remove commented code

File: src/main/java/com/chaos/invoicify/service/InvoiceService.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+                    );
+                }).collect(Collectors.toList());
+    }
+
+    public List<ItemDto> addItems(String invoiceName, List<ItemDto> itemDtoList) {
+        InvoiceEntity invoiceEntity = this.invoicesRepository.findByInvoiceName(invoiceName);
+        if (invoiceEntity != null) {
+            System.out.println("Invoice inside addItems: " + invoiceEntity.getInvoiceName() + " " + invoiceEntity.getCompanyName());
+        } else {
+            System.out.println("returned a null");
```
Comment (samik-saha) [inline src/main/java/com/chaos/invoicify/service/InvoiceService.java]: Remove system println methods.

File: src/main/java/com/chaos/invoicify/service/InvoiceService.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        }
+
+        itemDtoList
+            .stream()
+            .forEach(itemDto -> {
+                System.out.println(itemDto.getItemDescription());
```
Comment (samik-saha) [inline src/main/java/com/chaos/invoicify/service/InvoiceService.java]: remove println

File: src/test/java/com/chaos/invoicify/Invoices/InvoiceIT.java [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("length()").value(1))
+            .andExpect(jsonPath("[0].invoiceName").value("Invoice1"))
+            .andExpect(jsonPath("[0].companyName").value("Company1"))
+            .andExpect(jsonPath("[0].invoiceDate").value("2021-05-08"))
+            .andExpect(jsonPath("[0].itemDtoList").isEmpty())
+            .andDo(document("GetInvoices", responseFields(
+                fieldWithPath("[0].invoiceName").description("Invoice1")
+                , fieldWithPath("[0].companyName").description("Company1")
+                , fieldWithPath("[0].invoiceDate").description("2021-05-08")
```
Comment (samik-saha) [inline src/test/java/com/chaos/invoicify/Invoices/InvoiceIT.java]: revise the description

**Review by samik-saha — CHANGES_REQUESTED** [review]
Unit test cases for service are missing

File: src/main/java/com/chaos/invoicify/dto/InvoiceDto.java (hunk 1 above)
Comment (vinay-py) [reply src/main/java/com/chaos/invoicify/dto/InvoiceDto.java]: Done

File: src/main/java/com/chaos/invoicify/entity/InvoiceEntity.java (hunk 2 above)
Comment (vinay-py) [reply src/main/java/com/chaos/invoicify/entity/InvoiceEntity.java]: Done

File: src/main/java/com/chaos/invoicify/service/InvoiceService.java (hunk 3 above)
Comment (vinay-py) [reply src/main/java/com/chaos/invoicify/service/InvoiceService.java]: Done

File: src/main/java/com/chaos/invoicify/service/InvoiceService.java (hunk 4 above)
Comment (vinay-py) [reply src/main/java/com/chaos/invoicify/service/InvoiceService.java]: Done

File: src/main/java/com/chaos/invoicify/service/InvoiceService.java (hunk 5 above)
Comment (vinay-py) [reply src/main/java/com/chaos/invoicify/service/InvoiceService.java]: Done

File: src/test/java/com/chaos/invoicify/Invoices/InvoiceIT.java (hunk 6 above)
Comment (vinay-py) [reply src/test/java/com/chaos/invoicify/Invoices/InvoiceIT.java]: Done

**Review by samik-saha — APPROVED** [review]
We can merge it with main

**Review by vinay-py — APPROVED** [review]

**Pull request merged**